bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/items/MetadataConfigDescriptionProvider.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.config.core.items;
+
+import java.util.List;
+import java.util.Locale;
+
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.core.ConfigDescriptionParameter;
+import org.eclipse.smarthome.config.core.ParameterOption;
+
+/**
+ *
+ * @author Simon Kaufmann - initial contribution and API
+ *
+ */
+public interface MetadataConfigDescriptionProvider {
+
+    /**
+     * Get the identifier of the metadata namespace
+     *
+     * @return the metadata namespace
+     */
+    String getNamespace();
+
+    /**
+     * Get the human-readable description of the metadata namespace
+     * <p>
+     * Overriding this method is optional - it will default to the namespace identifier.
+     *
+     * @param locale a locale, if available
+     * @return the metadata namespace description
+     */
+    default String getDescription(@Nullable Locale locale) {
+        return getNamespace();
```

Comment:
I would remove the default implementation as people won't notice that they could/should overwrite it and you have already implemented the fallback [here](https://github.com/eclipse/smarthome/pull/4390/commits/5924264500e248d386d29a2802a4cb3db88e17c4#diff-152fa07bd689237e8513d6766d765f0bR122) as well - one place should be enough and this method should be allowed to return null.